horovod/mxnet/tensor_util.cc

```diff
@@ -137,13 +138,14 @@ void TensorUtil::Free(NDArray* tensor) { delete tensor; }
 
 // Resize tensor to nDimension with length size[i] in dimension i
 void TensorUtil::ResizeNd(NDArray* tensor, int nDimension, int64_t* size) {
-  void* temp_out;
-  MXNDArrayReshape64(tensor, nDimension, size, false, &temp_out);
-  tensor = static_cast<NDArray*>(temp_out);
+  mxnet::Tuple<dim_t> shape(size, size+nDimension);
+  tensor->ReshapeAndAlloc(shape);
```

Comment:
Can we still use the C API MXNDArrayReshape64 here instead of calling the lower level method inside NDArray?

Reply:
After discussion, `MXNDArrayReshape64` is not a fit here. We should find a correct C API function for reshaping with a different size.